### ISSUE TITLE
@iliaschalkidis I didn't dig into this far. However from my experience with Keras, "causal" means the output at time step t is only related to outputs from 0 to (t-1). Did you mean if I choose kernel_size=3 with "causual", as the padding is transfered to "valid" by tensorflow, the output at t is not related to the outputs from 0 to (t-1) ? or the output is no longer a sequence output?

### ISSUE THREAD

**Issue by liangstein** [liangstein/ByteNet-Keras#2]
_Originally posted by @liangstein in https://github.com/liangstein/ByteNet-Keras/issue_comments#issuecomment-425307037_
